tests/cupy_tests/manipulation_tests/test_kind.py

```diff
@@ -10,6 +10,13 @@
 @testing.gpu
 class TestKind(unittest.TestCase):
 
+    @testing.for_all_dtypes()
+    def test_asarray_chkfinite(self, dtype):
```

Comment:
Could you add another test with a non-finite-value array? I am not sure if bool/int/complex types make sense here, so worth a look.

Reply:
Hi, @leofang! 
I added the tests for the non-finite-value array, and all the tests are passed in the CI. Are more tests required?
> I am not sure if bool/int/complex types make sense here, so worth a look.

So, why did you think that was worth checking?
I checked for `numpy.asarray_chkfinite`. The following were the outputs: (which are as expected)

#### For bool
```
In [12]: a = numpy.asarray_chkfinite([numpy.inf, 0., -numpy.inf], dtype=bool)

In [13]: a
Out[13]: array([ True, False,  True])
```
#### For int
```
In [14]: a = numpy.asarray_chkfinite([numpy.inf, 0., -numpy.inf], dtype=int)
---------------------------------------------------------------------------
OverflowError                             Traceback (most recent call last)
<ipython-input-14-d6d91c379cab> in <module>
----> 1 a = numpy.asarray_chkfinite([numpy.inf, 0., -numpy.inf], dtype=int)

~/anaconda3/lib/python3.8/site-packages/numpy/lib/function_base.py in asarray_chkfinite(a, dtype, order)
    481 
    482     """
--> 483     a = asarray(a, dtype=dtype, order=order)
    484     if a.dtype.char in typecodes['AllFloat'] and not np.isfinite(a).all():
    485         raise ValueError(

~/anaconda3/lib/python3.8/site-packages/numpy/core/_asarray.py in asarray(a, dtype, order)
     81 
     82     """
---> 83     return array(a, dtype, copy=False, order=order)
     84 
     85 

OverflowError: cannot convert float infinity to integer
```
#### For float
```
In [15]: a = numpy.asarray_chkfinite([numpy.inf, 0., -numpy.inf], dtype=float)
---------------------------------------------------------------------------
ValueError                                Traceback (most recent call last)
<ipython-input-15-15536ad3bf83> in <module>
----> 1 a = numpy.asarray_chkfinite([numpy.inf, 0., -numpy.inf], dtype=float)

~/anaconda3/lib/python3.8/site-packages/numpy/lib/function_base.py in asarray_chkfinite(a, dtype, order)
    483     a = asarray(a, dtype=dtype, order=order)
    484     if a.dtype.char in typecodes['AllFloat'] and not np.isfinite(a).all():
--> 485         raise ValueError(
    486             "array must not contain infs or NaNs")
    487     return a

ValueError: array must not contain infs or NaNs
```
#### For Complex
```
a = numpy.asarray_chkfinite([numpy.inf, 0., -numpy.inf], dtype=complex)
---------------------------------------------------------------------------
ValueError                                Traceback (most recent call last)
<ipython-input-16-14e1c1a2ac61> in <module>
----> 1 a = numpy.asarray_chkfinite([numpy.inf, 0., -numpy.inf], dtype=complex)

~/anaconda3/lib/python3.8/site-packages/numpy/lib/function_base.py in asarray_chkfinite(a, dtype, order)
    483     a = asarray(a, dtype=dtype, order=order)
    484     if a.dtype.char in typecodes['AllFloat'] and not np.isfinite(a).all():
--> 485         raise ValueError(
    486             "array must not contain infs or NaNs")
    487     return a

ValueError: array must not contain infs or NaNs
```

Reply:
> and all the tests are passed in the CI

Just FYI: actual GPU tests are not yet run, the green CIs are just preliminary checks (formatting, compilation, etc).

> > I am not sure if bool/int/complex types make sense here, so worth a look.
> 
> So, why did you think that was worth checking? 

Because experience showed that these are often corner cases in newly added APIs that require special treatments. It'd be nice to take care of them in one shot.

> I checked for `numpy.asarray_chkfinite`. The following were the outputs: (which are as expected)

Thank you, @khushi-411. Looks like the current implementation should work fine. I'll make a few minor, style suggestions shortly.

Reply:
> actual GPU tests are not yet run

Oo, that's the reason (: 
> It'd be nice to take care of them in one shot.

Ok, I understood. I'll try to figure out those cases the next time.